tracer/src/Datadog.Trace/Configuration/ConfigurationSources/JsonConfigurationSource.cs

```diff
@@ -30,7 +32,9 @@ public class JsonConfigurationSource : IConfigurationSource
         /// <param name="json">A JSON string that contains configuration values.</param>
         public JsonConfigurationSource(string json)
```

Comment:
Shouldn't it be `public JsonConfigurationSource(string? json)`? 🤔

Reply:
I don't think so. In reality, we have only ever accepted non-null strings here (because `JsonConvert.DeserializeObject(json)` throws if you pass it `null`. So leaving it as `non-null` (and adding the explicit check) matches the existing behavior, gives the correct contract, and gives a better error message now 🙂